src/test-helpers.tsx

```diff
@@ -80,3 +81,54 @@ export const createContextWithFakeRouter = ({
     ...overrides,
   };
 };
+
+/*
+ * Repeatedly render a component tree using enzyme.shallow() until
+ * finding and rendering TargetComponent.
+ *
+ * This is useful for testing a component wrapped in one or more
+ * HOCs (higher order components).
+ *
+ * The `componentInstance` parameter is a React component instance.
+ * Example: <MyComponent {...props} />
+ *
+ * The `targetComponent` parameter is the React class (or function) that
+ * you want to retrieve from the component tree.
+ */
+type ShallowUntilTargetParams = {
+  componentInstance: React.ReactElement<any>;
+  targetComponent: React.JSXElementConstructor<any>;
+  options?: {
+    maxTries?: number;
+    shallowOptions?: object;
+    _shallow?: typeof shallow;
+  };
+};
+
+export const shallowUntilTarget = ({
+  componentInstance,
+  targetComponent,
+  options: { maxTries = 10, shallowOptions = {}, _shallow = shallow } = {},
```

Comment:
We don't want to assign an empty object here because that does not make much sense. It will fail if we don't specify some arguments, so I'd like to see the old signature (only the 3rd parameter is an object).